### ISSUE TITLE
Not getting IP-address

### ISSUE THREAD

**Issue by PaulskPt** [sandeepmistry/pico-rmii-ethernet#2]
Hi, @sandeepmistry 
In the first place I want to thank you for making available this repo. 

I am using a RPi Pico that came soldered onto a Cytron Maker Pi Pico board (that I bought from: 'The Pi Hut' U.K. (see:  https://thepihut.com/products/maker-pi-pico-with-pre-soldered-pico). 
To ithis board  I connected a RMII Ethernet PHY module that I bought from Pimoroni Ltd. (see: https://shop.pimoroni.com/products/rmii-ethernet-phy-breakout) (ref. BRK0001). 

I connected the PHY module according to the instructions described in: https://www.raspberrypi.com/news/how-to-add-ethernet-to-raspberry-pi-pico/. 
I git cloned your repo. Followed your instructions. The source built OK without any error. 
Then I put the Pico in BOOTSEL mode. I copied the file: pico_rmii_ethernet_httpd.uf2 onto the RPI-RP2 disk.

The PHY modules is connected to the Local Area Network in my house, using a UTP-cable connected to a 16-port D-Link switch that is connected to the central router of my ISP. The UTP cable is OK. I used it with succes to connect a RPi-400 to the internet.
The length of the UTP cable is about 2 m.

Observations:
a) the PHY module: 
The amber led on the PHY module is on; the green led is blinking at times. 

b) Maker Pi Pico board:
I observed the following due to the blinking or not of the leds connected to the used GPIO ports.

![2021-11-14_18h13_GPIO_Func_LEDstatus](https://user-images.githubusercontent.com/9107950/141693332-bc00d4d8-4a3e-4b4e-a42d-34d7e0571365.png)

When I disconnect the lines from the PHY module to GPIO pins 15 and 20, the LEDs of these ports stay ON.

The following ports on the Maker Pi Pico board are not connected to the PHY module, but the LEDs are ON nevertheless.
21, 22         |   constantly ON

c) When running my PC in MS Windows I used the WireShark app to see if I could discover transmissions from or to the RPi Pico with PHY module. I did not discover any activity.

Initially I built the .uf2 file during a WSL session on a Windows 10 O.S.
Secondly I built the .uf2 file from within an Ubuntu 20.04 O.S. session.

The wires that connect the GPIO pins, GND and 3V3 pins from the Maker Pi Pico board to the PHY module
are brand new (each about 19cm of length).

So far the description of what I use; what I did and what I observed thus far.

Question: 
Do you have any idea what can cause the fact that this setup cannot obtain an IP-address from the router ?

Attachment: .uf2 run output to CLI during an Ubuntu LInux session:

Sometimes the output report: 'Netif link status changed down'.

![2021-11-14_18h00_RPi_Pico_w_PHY_module_output](https://user-images.githubusercontent.com/9107950/141692942-1f2bb360-4681-4a16-a201-ff6c400b81b1.png)

Greeting,
Paulus

**Comment by MicroKoder** [comment]
i have same issue, the only messages:
pico rmii ethernet - httpd
netif status changed 0.0.0.0
and nothing more

**Comment by MicroKoder** [comment]
I got solution for my case. I have DP83848 module, it need pull-up on MDIO pin, after adding it all started.
I suggest you check logs in the router

**Comment by PaulskPt** [comment]
Hi @MicroKoder, thank you for sharing this. I am going to try it some day later. I am now 'fighting' with some other device :-)

**Comment by PaulskPt** [comment]
I got it working now. I re-wired the setup as described in  https://www.raspberrypi.com/news/how-to-add-ethernet-to-raspberry-pi-pico/. First checking the wire definitions in: pico-rmii-ethernet\src\include\rmii_ethernet\netif.h. The CMake, make and flash process went swift. On a RPi400 running minicom I received the netif status and it gave a local IP address from inside the LAN. Then on my desktop PC running a browser I visited the server page of the Pico that showed info about Lwip as described in the article 'how-to-add-ethernet-to-raspberry-pi-pico'.  I did not need any pullup resistor.
See the image of the hardware below.

![2021-12-11_22h33_RPi_Pico_and_RMII_Eth_module](https://user-images.githubusercontent.com/9107950/145694079-e049338f-805b-4f2f-875f-0e1e9576f661.jpg)
